android/src/main/kotlin/com/situm/situm_flutter/Extensions.kt

```diff
@@ -96,6 +97,15 @@ fun LocationRequest.Builder.fromArguments(args: Map<String, Any>): LocationReque
         }
         outdoorLocationOptions(outdoorLocationOptions.build())
     }
+    if (args.containsKey("diagnosticsOptions")) {
+        val diagnosticsOptionsMap = args["diagnosticsOptions"] as Map<String, Any>
+        val diagnosticsOptions = DiagnosticsOptions.fromMap(diagnosticsOptionsMap)
+        diagnosticsOptions(diagnosticsOptions)
```

Comment:
Tontería = resultoume algo lioso leer esta liña, parece que estas metendo un objeto dentro de sí mismo. Un `diagnosticsOptions(dgOptions)` quedaría máis claro